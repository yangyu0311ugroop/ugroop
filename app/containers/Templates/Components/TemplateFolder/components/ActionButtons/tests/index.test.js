import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import { ActionButtons } from '../index';

describe('<ActionButtons />', () => {
  let rendered;
  let instance;

  const resaga = {
    setValue: jest.fn(),
    dispatchTo: jest.fn(),
  };

  const props = {
    classes: {},
    resaga,
    onCopy: jest.fn(),
    onMove: jest.fn(),
    onEdit: jest.fn(),
    onDelete: jest.fn(),
    canMove: true,
  };

  beforeEach(() => {
    rendered = shallow(<ActionButtons {...props} />);
    instance = rendered.instance();
  });

  it('should exists', () => {
    expect(ActionButtons).toBeDefined();
  });

  it('should render without exploding', () => {
    expect(rendered.length).toBe(1);
  });

  describe('handleClickMenu()', () => {
    it('should set anchorEl state to the currentTarget', () => {
      instance.handleClickMenu({ currentTarget: 'Jesus' });
      expect(rendered.state().anchorEl).toBe('Jesus');
    });
  });

  describe('handleCloseMenu()', () => {
    it('should set anchorEl state to null', () => {
      rendered.setState({ anchorEl: 'World' });
      instance.handleCloseMenu();
      expect(rendered.state().anchorEl).toBe(null);
    });
  });

  describe('handleEvents()', () => {
    it('should set anchorEl state to null before calling the event function', () => {
      rendered.setState({
        anchorEl: 'something',
      });
      const func = jest.fn();
      instance.handleEvents(func)();
      expect(rendered.state().anchorEl).toBe(null);
      expect(func).toBeCalled();
    });
  });

  describe('render()', () => {
    it('should render correctly', () => {
      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });
});
