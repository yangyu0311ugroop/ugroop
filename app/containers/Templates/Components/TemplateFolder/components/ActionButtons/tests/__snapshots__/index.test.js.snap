// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButtons /> render() should render correctly 1`] = `
<div>
  <WithStyles(Button)
    color="gray"
    icon="ellipsis"
    iconButton={true}
    onClick={[Function]}
    size="small"
    variant="borderless"
  />
  <WithStyles(ForwardRef(Menu))
    anchorEl={null}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(MenuItem))
      button={true}
      onClick={[Function]}
    >
      <WithStyles(Icon)
        icon="pencil3"
      />
      <FormattedMessage
        defaultMessage="Edit"
        id="app.containers.Templates.Components.TemplateFolder.editLabel"
      />
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      button={true}
      onClick={[Function]}
    >
      <WithStyles(Icon)
        icon="trash2"
      />
      <FormattedMessage
        defaultMessage="Delete"
        id="app.containers.Templates.Components.TemplateFolder.deleteLabel"
      />
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      button={true}
      onClick={[Function]}
    >
      <WithStyles(Icon)
        icon="compare"
      />
      <FormattedMessage
        defaultMessage="Move"
        id="app.containers.Templates.Components.TemplateFolder.moveLabel"
      />
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      button={true}
      onClick={[Function]}
    >
      <WithStyles(Icon)
        icon="copy"
      />
      <FormattedMessage
        defaultMessage="Copy"
        id="app.containers.Templates.Components.TemplateFolder.copyLabel"
      />
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Menu))>
</div>
`;
